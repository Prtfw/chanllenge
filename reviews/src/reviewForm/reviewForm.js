import React from "react";
import { withRouter } from "react-router-dom";

export const mapIdToEmployeeName = (id, employees) => {
  return employees && employees.filter(emp => emp.guid === id)[0];
};

class ReviewForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { score: 5, comment: "" };
  }

  componentDidMount() {
    this.getEmployees();
    this.getReviews();
  }
  getReviews = () =>
    fetch("http://localhost:3008/reviews")
      .then(res => res.json())
      .then(data => this.setState({ reviews: data }));
  getEmployees = () =>
    fetch("http://localhost:3008/employees")
      .then(res => res.json())
      .then(data => this.setState({ employees: data }));

  handleSubmit = evt => {
    evt.preventDefault();
    fetch(`http://localhost:3008/submitReview/${this.props.match.params.id}`, {
      method: "post",
      headers: {
        Accept: "application/json, text/plain, */*",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ ...this.state, id: this.props.match.params.id }),
    }).then(res => this.props.history.push("/"))
  };
  render() {
    return (
      <div>
        {this.props.match.params.for ? (
          <div>
            Review for{" "}
            {`${(mapIdToEmployeeName(this.props.match.params.for, this.state.employees) || {})
              .first}, ${(mapIdToEmployeeName(this.props.match.params.for, this.state.employees) ||
              {}).last}`}
          </div>
        ) : null}
        <form action="">
          <label> least awsome </label>
          <input
            type="range"
            min="0"
            max="5"
            step="1"
            onChange={e => this.setState({ score: e.target.value })}
          />{" "}
          <label> most awsome, score: {this.state.score} </label>
          <textarea type="textarea" onChange={e => this.setState({ comment: e.target.value })}>
            why? your comments here.
          </textarea>
          <input type="submit" onClick={this.handleSubmit} />
        </form>
      </div>
    );
  }
}

export default withRouter(ReviewForm);
