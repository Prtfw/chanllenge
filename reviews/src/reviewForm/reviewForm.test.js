// example of a test for a pure helper function using snapshot testing

import { mapIdToEmployeeName } from "./reviewForm";

const employeesDataSeed = [
  {
    first: "Stella",
    last: "Ellis",
    guid: "d6ee9507-6a7a-5d89-a92b-e8fc13c0e2f8",
  },
  {
    first: "Gavin",
    last: "Schwartz",
    guid: "98fce8dd-7420-5dad-b315-101e1ebd18c0",
  },
  {
    first: "Derek",
    last: "Walker",
    guid: "efcda8fc-a1d0-5c20-92bb-9e96d4abcf08",
  },
  {
    first: "Della",
    last: "Wagner",
    guid: "50535380-f046-5d50-acf9-8285752777b6",
  },
];

describe("mapIdToEmployeeName()", () => {
  test("Finds Employee by ID", () => {
    const { guid } = mapIdToEmployeeName("d6ee9507-6a7a-5d89-a92b-e8fc13c0e2f8", employeesDataSeed);
    expect(guid).toMatchSnapshot() // use snapshot testing after verifying result by using line below
    //toBe("d6ee9507-6a7a-5d89-a92b-e8fc13c0e2f8");
  });
});
