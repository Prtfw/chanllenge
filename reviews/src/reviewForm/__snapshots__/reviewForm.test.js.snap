// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapIdToEmployeeName() Finds Employee by ID 1`] = `"d6ee9507-6a7a-5d89-a92b-e8fc13c0e2f8"`;
